File: test_db_state.py
import urllib.request
import json
import re

with open('js/supabase.js', 'r') as f:
    content = f.read()
    match = re.search(r"const\s+supabaseKey\s*=\s*'([^']+)'", content)
    if match:
        key = match.group(1)
        url = "https://gipxxcfydceahzmqdoks.supabase.co/rest/v1/trades?user_id=eq.66&select=id,status,outstanding_amount,paid_amount,symbol,total_amount&order=created_at.desc"
        req = urllib.request.Request(url, headers={
            'apikey': key,
            'Authorization': f'Bearer {key}'
        })
        try:
            with urllib.request.urlopen(req) as response:
                data = json.loads(response.read().decode())
                print(json.dumps(data, indent=2))
        except Exception as e:
            print("Error:", e)
    else:
        print("Key not found")

File: test_db_state.js
const fs = require('fs');
const https = require('https');

const supabaseKey = fs.readFileSync('js/supabase.js', 'utf8').match(/const supabaseKey = '(.*?)'/)[1];
const supabaseUrl = 'https://xizuwvmepfcfodwfwqce.supabase.co';

const headers = {
    'apikey': supabaseKey,
    'Authorization': `Bearer ${supabaseKey}`,
    'Content-Type': 'application/json'
};

const req = https.request(`${supabaseUrl}/rest/v1/trades?user_id=eq.66&select=id,status,outstanding_amount,paid_amount,symbol,total_amount&order=created_at.desc`, { headers }, (res) => {
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => console.log(JSON.parse(data)));
});
req.on('error', console.error);
req.end();
